storage/sqlite/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate core;
+
+mod error;
+
+use rusqlite::{Connection, OpenFlags, OptionalExtension};
+use std::borrow::Cow;
+use std::cmp::max;
+use std::path::PathBuf;
+use std::sync::{Mutex, MutexGuard};
+use std::vec::IntoIter;
+
+use crate::error::process_sqlite_error;
+use storage_core::{
+    backend::{self, TransactionalRo, TransactionalRw},
+    info::DbDesc,
+    Data, DbIndex,
+};
+use utils::shallow_clone::ShallowClone;
+use utils::sync::Arc;
+
+/// Return the table name that corresponds to a specific index
+/// TODO this is unoptimised and the returned values should be cached
+fn db_table_name(idx: usize) -> String {
```

Comment:
Since this is only used with database indices, it would be more appropriate to take `DbIndex` rather than plain `usize` to make the types more precise. (Also spares you a bunch of `.get()` elsewhere)

Reply:
Removed this